.github/scripts/trymerge.py

```diff
@@ -1060,34 +1074,43 @@ def find_matching_merge_rule(pr: GitHubPR,
         if len(approvers_intersection) == 0 and len(rule_approvers_set) > 0:
             if reject_reason_score < 10000:
                 reject_reason_score = 10000
-                reject_reason = (f"Matched rule {rule_name}, but PR #{pr.pr_num} was not reviewed yet by any of: " +
-                                 f"{', '.join(list(rule_approvers_set)[:5])}{', ...' if len(rule_approvers_set) > 5 else ''}")
+                reject_reason = "\n".join((
+                    f"Approval needed from one of the following (Rule '{rule_name}'):",
+                    f"{', '.join(list(rule_approvers_set)[:5])}{', ...' if len(rule_approvers_set) > 5 else ''}"
+                ))
             continue
+
+        # Does the PR pass the checks required by this rule?
         mandatory_checks = rule.mandatory_checks_name if rule.mandatory_checks_name is not None else []
         checks = get_combined_checks_from_pr_and_land_validation(pr, land_check_commit)
         required_checks = filter(lambda x: force is False or "CLA Check" in x, mandatory_checks)
         [pending_checks, failed_checks] = categorize_checks(checks, required_checks)
 
+        hud_link = f"https://hud.pytorch.org/{pr.org}/{pr.project}/commit/{pr.last_commit()['oid']}"
         if len(failed_checks) > 0:
             if reject_reason_score < 30000:
                 reject_reason_score = 30000
-                reject_reason = (
-                    f"[View failures on hud](https://hud.pytorch.org/{pr.org}/{pr.project}/commit/{pr.last_commit()['oid']}). "
-                    + f"Refusing to merge as mandatory check(s) {checks_to_str(failed_checks)} failed for "
-                    + f"rule {rule_name}."
-                )
+                reject_reason = "\n".join((
+                    f"The following mandatory check(s) failed (Rule `{rule_name}`):"
+                    f"{checks_to_str(failed_checks)}",
```

Comment:
I'm getting a bit greedy here, but do you think it's better if we promote HUD link here?  I'm thinking to replace the link in the failed checks with HUD link.  Taking https://github.com/pytorch/pytorch/pull/84035#issuecomment-1226909106 as an example, the message will be as follows with pull check pointing to HUD:

> Refusing to merge as mandatory check(s) [pull](https://hud.pytorch.org/pytorch/pytorch/commit/28551516539442a374ea06048e63c2022be7c097) failed for rule superuser.

We could add some anchors to the HUD page for different checks later on.  IMO, this gives people a little nudge to go to HUD instead of jumping to GitHub

Reply:
100%. I'm thinking we need anchor links on HUD first though or something that'll let us take the user to a link that's different from the default hud page (which is already linked to in the "Debug by viewing the failures on HUD" line).  

Otherwise all the links would point to the same place

Reply:
Actually, the current pull link points to the whole entire pull workflow, right? Not a specific failed job? We could just replace pull with the link to the HUD PR page, no?

Reply:
That would be a strict improvement, I think

Reply:
We could...and you know what, I agree that's better.  Okay, I'm going to change the link and remove the "Debug by viewing on HUD line" since it'll become redundant

Reply:
Ready for round N+1 of reviews :)

Reply:
> I'm thinking we need anchor links on HUD first though

You could just append the soon-to-be-created anchor I think, i.e. `HUD_LINK#pull` if you prefer to have that right away. I don't think the browser will complain. If the anchor doesn't exist, the page is loaded as usual.

Reply:
Is someone working on those anchors?

Reply:
@ZainRizvi Here you do https://github.com/pytorch/test-infra/pull/650, please take a look